### PR TITLE
JCLOUDS-664:Implement OSImageToImage Function

### PR DESCRIPTION
**Pull request by hsbhathiya**
implement OSImageToImage Function.
Changes to API implementation
1- Added PublisherName to OSImage
2-Changed Locations list to Location

### CODE REVIEW

**Comment by hsbhathiya** [comment]
@andreaturli , @nacx 
I've addressed most of  the review suggestions. 
@nacx 
Not clear about how Image.Status is used

File: azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java [hunk 1]
```diff
@@ -16,15 +16,121 @@
...
+   private static final String COREOS = "CoreOS";
+   private static final String OPENSUSE = "openSUSE";
+   private static final String SUSE = "SUSE";
+   private static final String SQL_SERVER = "SQL Server";
+   private static final String ORACLE_lINUX = "Orcale Linux";
+
+   private final JustProvider provider;
+
+   @Inject
+   protected OSImageToImage(JustProvider provider) {
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java]: Make it package private by removing the `protected` modifier.

File: azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java [hunk 2]
```diff
@@ -16,15 +16,121 @@
...
+         @Override
+         public OperatingSystem.Builder apply(final OSImage image) {
+
+            final String label = image.label();
+            checkNotNull(label, "label");
+            OsFamily family = OsFamily.UNRECOGNIZED;
+
+            if (label.contains(CENTOS))
+               family = OsFamily.CENTOS;
+            else if (label.toUpperCase().contains(OPENLOGIC))
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java]: I'm a bit confused. Why the label is compared upper case in some cases and not everywhere?

**Comment by nacx** [comment]
Just a couple comments. If @andreaturli does not say the opposite this lgtm. Thanks!

**Comment by hsbhathiya** [comment]
@andreaturli , @nacx 
Shall we get this merged?

**Pull request merged**